src/main/java/gyro/google/compute/AbstractForwardingRuleResource.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.api.services.compute.model.ForwardingRule;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.google.Copyable;
+
+public abstract class AbstractForwardingRuleResource extends ComputeResource implements Copyable<ForwardingRule> {
+
+    private String ipAddress;
+    private String ipProtocol;
+    private Boolean allPorts;
+    private String description;
+    private String ipVersion;
+    private String loadBalancingScheme;
+    private String name;
+    private String networkTier;
+    private String portRange;
+    private List<String> ports;
+    private String serviceLabel;
+
+    // Read-only
+    private String selfLink;
+    private String serviceName;
+
+    /**
+     * IP address that this forwarding rule serves.
+     */
+    public String getIpAddress() {
+        return ipAddress;
+    }
+
+    public void setIpAddress(String ipAddress) {
+        this.ipAddress = ipAddress;
+    }
+
+    /**
+     * The IP protocol for the forwarding rule. Valid values are ``TCP``, ``UDP``, ``ESP``, ``AH``, ``SCTP`` or ``ICMP``.
+     */
+    @Required
+    @ValidStrings({ "TCP", "UDP", "ESP", "AH", "SCTP", "ICMP" })
+    public String getIpProtocol() {
+        return ipProtocol;
+    }
+
+    public void setIpProtocol(String ipProtocol) {
+        this.ipProtocol = ipProtocol;
+    }
+
+    /**
+     * Enabling this allows packets to be forwarded to the backends configured with this forwarding rule over all the configured ports.
+     */
+    public Boolean getAllPorts() {
+        return allPorts;
+    }
+
+    public void setAllPorts(Boolean allPorts) {
+        this.allPorts = allPorts;
+    }
+
+    /**
+     * Description of the forwarding rule.
+     */
+    @Updatable
```

Comment:
Even though this is marked updatable, the update method does not update this field.